packages/react-devtools-shared/src/devtools/views/Components/KeyValue.js

```diff
@@ -330,9 +336,9 @@ export default function KeyValue({
           key={index}
           alphaSort={alphaSort}
           bridge={bridge}
-          canDeletePaths={canDeletePaths && !isReadOnly}
-          canEditValues={canEditValues && !isReadOnly}
-          canRenamePaths={canRenamePaths && !isReadOnly}
+          canDeletePaths={canDeletePathsProp && !isReadOnly}
+          canEditValues={canEditValuesProp && !isReadOnly}
+          canRenamePaths={canRenamePathsProp && !isReadOnly}
```

Comment:
nit: Same comment as above. I don't think it makes sense (conceptually) to disallow editing of an outer `<KeyValue>` but then allow editing in an inner `<KeyValue>` when `readOnly === false` so we can just make these:

```suggestion
          canDeletePaths={canDeletePaths}
          canEditValues={canEditValues}
          canRenamePaths={canRenamePaths}
```

Reply:
`isReadOnly` is different from the `readOnly` variable that I introduced (perhaps I should rename `readOnly` to `readOnlyFlag`?). And I think `isReadOnly` could be `false` while `canEditValues` is `true`.
Do you simply want me to remove `canEditValuesProp` etc. and use `canEditValues` etc. instead? So this would read `canEditValues={canEditValues && !isReadOnly}` ?

Reply:
> `isReadOnly` is different from the `readOnly` variable that I introduced

Oh! Right. I read too quickly and overlooked that fact.

> perhaps I should rename `readOnly` to `readOnlyFlag`?

That sounds helpful. Maybe we could change `isReadOnly` too to something like `isReadOnlyBasedOnMetadata` 😅 Just to make it totally clear to someone like me who may fail to notice.

> Do you simply want me to remove `canEditValuesProp` etc. and use `canEditValues` etc. instead? So this would read `canEditValues={canEditValues && !isReadOnly}` ? 

Yeah. I think you should be passing e.g. `canDeletePaths` rather than `canDeletePathsProp` here though.

```js
canDeletePaths={canDeletePaths && !isReadOnly}
```

Right?

Reply:
Yeah, that's what I meant.